### PR TITLE
Key Access Lib updates

### PR DESCRIPTION
**Pull request by cworsnop-figure**
## Description

Update to support the latest version of `Originator-Key-Access-Lib` which move much of the key management into the library (and thus out of this repository). 

(note: expected to fail until https://github.com/provenance-io/originator-key-access-lib/pull/3/files is published)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/p8e-cee-api/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Unit Test Results` in the comment section below once CI passes

### CODE REVIEW

File: models/src/main/kotlin/io/provenance/api/models/account/KeyManagementConfig.kt [hunk 1]
```diff
@@ -1,7 +1,6 @@
 package io.provenance.api.models.account
 
 data class KeyManagementConfig(
-    val address: String,
-    val tokenPath: String,
     val plugin: String = "io.provenance.plugins.vault.VaultPlugin",
+    val pluginSpec: Any
```
Comment (dkneisly-figure) [inline models/src/main/kotlin/io/provenance/api/models/account/KeyManagementConfig.kt:5]: Only thing I don't love is this being an `Any` type

Comment (cworsnop-figure) [reply models/src/main/kotlin/io/provenance/api/models/account/KeyManagementConfig.kt]: fixed :D

File: service/src/main/kotlin/io/provenance/api/domain/cee/ContractService.kt [hunk 2]
```diff
@@ -14,7 +14,7 @@ import java.util.UUID
 
 interface ContractService {
     fun getContract(contractName: String): Class<out P8eContract>
-    fun <T : P8eContract> setupContract(client: Client, contractClass: Class<T>, records: Map<String, Message>, scopeUuid: UUID, sessionUuid: UUID? = null, participants: Map<Specifications.PartyType, Originator>? = null, scope: ScopeResponse? = null, scopeSpecification: String, audiences: Set<PublicKey>): Session
+    fun <T : P8eContract> setupContract(client: Client, contractClass: Class<T>, records: Map<String, Message>, scopeUuid: UUID, sessionUuid: UUID? = null, participants: Map<Specifications.PartyType, KeyEntity>? = null, scope: ScopeResponse? = null, scopeSpecification: String, audiences: Set<PublicKey>): Session
```
Comment (rpatel-figure) [inline service/src/main/kotlin/io/provenance/api/domain/cee/ContractService.kt:17]: 😱 Where's the linting at?
```suggestion
    fun <T : P8eContract> setupContract(
        client: Client, contractClass: Class<T>,
        records: Map<String, Message>,
        scopeUuid: UUID,
        sessionUuid: UUID? = null,
        participants: Map<Specifications.PartyType, KeyEntity>? = null,
        scope: ScopeResponse? = null,
        scopeSpecification: String,
        audiences: Set<PublicKey>,
    ): Session
```

File: service/src/main/kotlin/io/provenance/api/domain/usecase/provenance/account/GetSigner.kt [hunk 3]
```diff
@@ -22,25 +17,17 @@ class GetSigner(
     private val entityManager: EntityManager,
     private val provenanceProperties: ProvenanceProperties,
 ) : AbstractUseCase<GetSignerRequest, Signer>() {
-    override suspend fun execute(args: GetSignerRequest): Signer {
-        val originator = entityManager.getEntity(KeyManagementConfigWrapper(args.uuid.toString(), args.account.keyManagementConfig))
+    override suspend fun execute(args: GetSignerRequest): Signer =
+        entityManager.getEntity(KeyManagementConfigWrapper(args.uuid.toString(), args.account.keyManagementConfig)).let {
```
Comment (rpatel-figure) [inline service/src/main/kotlin/io/provenance/api/domain/usecase/provenance/account/GetSigner.kt:21]: Nit: could you rename `it` here so this is easier to follow?

File: service/src/main/kotlin/io/provenance/api/domain/usecase/objectStore/store/models/StoreObjectRequest.kt [hunk 4]
```diff
@@ -2,16 +2,14 @@ package io.provenance.api.domain.usecase.objectStore.store.models
...
+import io.provenance.scope.encryption.model.KeyRef
 
 data class StoreObjectRequest(
     val bytes: ByteArray,
     val type: String?,
     val objectStoreUrl: String,
     val useObjectStoreGateway: Boolean,
-    val publicKey: PublicKey,
-    val privateKey: PrivateKey,
+    val keyRef: KeyRef,
```
Comment (rpatel-figure) [inline service/src/main/kotlin/io/provenance/api/domain/usecase/objectStore/store/models/StoreObjectRequest.kt:12]: Should we label this PR as minor? Will other endpoint request bodies, like those for contract execution, be updated in the same fashion as well?

File: service/src/main/kotlin/io/provenance/api/domain/cee/ContractService.kt (hunk 2 above)
Comment (cworsnop-figure) [reply service/src/main/kotlin/io/provenance/api/domain/cee/ContractService.kt:17]: :hmmm: :truestory:

File: service/src/main/kotlin/io/provenance/api/domain/usecase/provenance/account/GetSigner.kt (hunk 3 above)
Comment (cworsnop-figure) [reply service/src/main/kotlin/io/provenance/api/domain/usecase/provenance/account/GetSigner.kt]: done

File: service/src/main/kotlin/io/provenance/api/domain/usecase/objectStore/store/models/StoreObjectRequest.kt (hunk 4 above)
Comment (cworsnop-figure) [reply service/src/main/kotlin/io/provenance/api/domain/usecase/objectStore/store/models/StoreObjectRequest.kt:12]: ya, that's a good idea. 

All of the updates needed are applied in this PR. Contract execution resolves all needed key material inside the usecase itself (unlike the store object request as you've pointed out). in this case, the callers of the StoreObject use case are for protos / files and each manually create the request object needed here.

**Review by rpatel-figure — APPROVED** [review]

**Pull request merged**